tools/provision

```diff
@@ -0,0 +1,15 @@
+#! /usr/bin/env bash
+
+echo "Setup: Setting up Virtual Env"
+
+BASEDIR=`dirname $0`/..
+
+if [ ! -d "$BASEDIR/env" ]; then
+    virtualenv $BASEDIR/env --no-site-packages
```

Comment:
If `env`'s use case is equivalent to `node_modules`, then it should be at `$BASEDIR`, but [pipenv](https://github.com/kennethreitz/pipenv) puts project-specific envs at `~/.local/share/virtualenvs/` with an autogenerated virtualenv name. I don't know why it is this way in pipenv, but I guess this has the advantage that no relative pathing is required, should the `activate` script need to be accessed by another user, e.g. `zulip` or a bot process.